coral/src/app/features/connectors/details/settings/ConnectorSettings.test.tsx

```diff
@@ -0,0 +1,651 @@
+
+  describe("enables user to delete a connector", () => {
+    const originalConsoleError = console.error;
+
+    beforeEach(() => {
+      console.error = jest.fn();
+      mockDeleteConnector.mockImplementation(jest.fn());
+      mockedUseConnectorDetails.mockReturnValue(mockConnectorDetails);
+
+      customRender(
+        <AquariumContext>
+          <ConnectorSettings />
+        </AquariumContext>,
+        {
+          memoryRouter: true,
+          queryClient: true,
+        }
+      );
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+      cleanup();
+      console.error = originalConsoleError;
+    });
+
+    it('shows a confirmation modal when user clicks "Request connector deletion"', async () => {
+      const confirmationModalBeforeClick = screen.queryByRole("dialog");
+      expect(confirmationModalBeforeClick).not.toBeInTheDocument();
+
+      const button = screen.getByRole("button", {
+        name: "Request connector deletion",
+      });
+
+      await user.click(button);
+      const confirmationModal = screen.getByRole("dialog");
+
+      expect(confirmationModal).toBeVisible();
+      expect(console.error).not.toHaveBeenCalled();
+    });
+
+    it('removes modal and does not Request connector deletion if user clicks "cancel"', async () => {
+      const button = screen.getByRole("button", {
+        name: "Request connector deletion",
+      });
+
+      await user.click(button);
+
+      const dialog = screen.getByRole("dialog");
+      expect(dialog).toBeVisible();
+
+      const cancelButton = within(dialog).getByRole("button", {
+        name: "Cancel",
+      });
+
+      await user.click(cancelButton);
+
+      expect(dialog).not.toBeInTheDocument();
+      expect(mockedNavigate).not.toHaveBeenCalled();
+      expect(mockDeleteConnector).not.toHaveBeenCalled();
+      expect(console.error).not.toHaveBeenCalled();
+    });
+
+    it("deletes connector successfully when user confirms deleting", async () => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      mockDeleteConnector.mockResolvedValue({ success: true });
+
+      const button = screen.getByRole("button", {
+        name: "Request connector deletion",
+      });
+
+      await user.click(button);
+
+      const dialog = screen.getByRole("dialog");
+      expect(dialog).toBeVisible();
+
+      const confirmButton = within(dialog).getByRole("button", {
+        name: "Request connector deletion",
+      });
+
+      await user.click(confirmButton);
+
+      expect(mockDeleteConnector).toHaveBeenCalledWith({
+        envId: testEnvironmentId,
+        connectorName: testConnectorName,
+        remark: undefined,
+      });
+      expect(mockedNavigate).toHaveBeenCalledWith("/connectors");
+      expect(dialog).not.toBeVisible();
+      expect(console.error).not.toHaveBeenCalled();
+    });
+
+    it("deletes connector and associates schemas successfully when user confirms deleting", async () => {
```

Comment:
That's an unnecessary test block, there's no schema to deleted in that case, that was only for topics. It's also the same flow like in the test before, so can be removed.